lib/charting/c3_charting.rb

```diff
@@ -0,0 +1,81 @@
+class C3Charting < Charting
+  # for Charting.detect_available_plugin
+  def self.available?
+    true
+  end
+
+  # for Charting.detect_available_plugin
+  def self.priority
+    0 # TODO set to value > JqplotCharting.priority to take effect
+  end
+
+  # backend identifier
+  def backend
+    :c3
+  end
+
+  # format for rails' render
+  def render_format
+    :json
+  end
+
+  # formatter for Rupport::Controller#render - see lib/report_formatter/...
+  def format
+    # TODO - the actual formatter
+    :c3
+  end
+
+  # called from each ApplicationController instance
+  def load_helpers(klass)
+    klass.instance_eval do
+      # TODO
+      helper C3Helper
+    end
+  end
+
+  def data_ok?(data)
+    !!YAML.load(data) && !YAML.load(data)[:options]
```

Comment:
`YAML.load` is not an extremely fast operation, so it is probably better to evaluate it once.